src/mlpack/core/tree/spill_tree/spill_tree_impl.hpp

```diff
@@ -674,8 +674,8 @@ bool SpillTree<MetricType, StatisticType, MatType, HyperplaneType, SplitType>::
   const double p1 = (double) (left + rightFrontier) / points.n_elem;
   const double p2 = (double) (right + leftFrontier) / points.n_elem;
 
-  if ((p1 <= rho || rightFrontier == 0) &&
-      (p2 <= rho || leftFrontier == 0))
+  if ((p1 <= this->NumChildren()*rho || rightFrontier == 0) &&
+      (p2 <= this->NumChildren()*rho || leftFrontier == 0) && tau != 0)
```

Comment:
Are you sure this is supposed to be a part of this PR?

Reply:
No,i was working on spill trees issues .i forgot to remove it .